src/adapters/mongoose.js

```diff
@@ -74,7 +74,9 @@ module.exports = (model, opts) => {
       }
 
       // NOTICE: Detect enum values for Enums in subdocument arrays.
-      if (fieldInfo.enum && fieldInfo.enum.length) {
+      if (fieldInfo.enum
+        && Array.isArray(fieldInfo.enum)
+        && fieldInfo.enumValues.length) {
         field.enums = fieldInfo.enum;
       }
```

Comment:
After some tests seen together, I think this part of code can simply be removed.
I've never managed to get a `fieldInfo.enum` which is an array.
It seems it is always the `enum validator` from:
- https://github.com/Automattic/mongoose/blob/master/lib/schema/number.js#L306
- https://github.com/Automattic/mongoose/blob/master/lib/schema/string.js#L204
- https://github.com/Automattic/mongoose/blob/master/lib/schema/array.js#L231
In the number case `fieldInfo.enum.length` will be `> 0` as the signature function takes 2 arguments (ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/length)

Furthermore, why `fieldInfo.enum.length` became `fieldInfo.enumValues.length`?

Reply:
It was a very interesting discussion indeed.
Removing this part of the code does not break tests & seems to still generate valid `.forestadmin-schema.json` (Even for type String with enum) so I definitely agree with the fact that this code doesn't seems to be useful anymore. 

However, I don't think our code coverage is enough to just remove it based on that, so I'll still dig just to be sure this is not used in some edge cases.

Thanks again for your deep dive into this !